Bearded.Utilities/Geometry/CircularArc2.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using OpenTK.Mathematics;
+
+namespace Bearded.Utilities.Geometry
+{
+    /// <summary>
+    /// Represents an arc of a circle between a pair of distinct points.
+    /// </summary>
+    public readonly struct CircularArc2 : IEquatable<CircularArc2>
+    {
+        public Vector2 Center { get; }
+        public float Radius { get; }
+        public Direction2 Start { get; }
+        public Angle Angle { get; }
+
+        public Direction2 End => Start + Angle;
+        public Vector2 StartPoint => Center + Radius * Start.Vector;
+        public Vector2 EndPoint => Center + Radius * End.Vector;
+
+        public float ArcLength => Angle.Radians * Radius;
+
+        /// <summary>
+        /// Returns the major arc if this arc is the minor arc, and returns the major arc if this is the minor arc.
+        /// </summary>
+        public CircularArc2 Opposite => new CircularArc2(Center, Radius, Start, oppositeAngle(Angle));
+
+        /// <summary>
+        /// Returns the same arc with the start and end directions swapped.
+        /// </summary>
+        public CircularArc2 Reversed => new CircularArc2(Center, Radius, End, -Angle);
+
+        /// <summary>
+        /// Constructs the minor arc in the circle with specified center and radius between the given directions.
+        /// </summary>
+        public static CircularArc2 ShortestArcBetweenDirections(
```

Comment:
This name makes me feel like maybe there's more than one possible Arc here. Would `ShorterArcBetweenDirections` be more correct? Or maybe even just `ShortArc..` or `MinorArc..` to use the mathematical term?

Reply:
I understand your point. I've renamed it to `MinorArc` now, but I want to explain my hesitation at not doing so originally. While minor and major arc are indeed the correct mathematical terms here, I think that most programmers to which this library would be aimed do not know. Even I wasn't aware that these were the terms until I ran into them. `Shortest` and `Longest` made more sense to me. `Shorter`... eh, I guess it works. `Short`... perhaps. LMK if you think using the mathematical term is a concern to you, and which alternative you prefer in that case.

Reply:
I'm happy using the mathematical term, however, I also had to look it up to be sure which is which. So... I'm not sure how helpful that is. Of all the non-mathematical ones 'Short' seems the best because it is concise, relatively ambiguous, and makes for good prose "get me the short arc between these directions".

Btw, does Direction2 have two ways of creating an angle from two directions and if so what are they called?

Reply:
`Short` and `Long` it is then.

It doesn't seem like `Angle` has a way to get the long angle between directions, it always return the shortest one. This is called `Angle.Between`. There are also `BetweenPositive` and `BetweenNegative` which always return a positive and negative angle respectively.